### ISSUE TITLE
Autocorrection of `Style/HashEachMethods` can break code

### ISSUE THREAD

**Issue by lzettel** [rubocop/rubocop#13318]
## Description

Adding items to a hash while looping through the hash itself results in an error in ruby: `can't add a new key into hash during iteration (RuntimeError)`.
But in the case ...
```ruby
x = { a: 1 }
x.keys.each do |key|
  x["#{key}_dup"] = 2
end
```
... rubocop wants to autocorrects this code to ...
```ruby
x = { a: 1 }
x.each_key do |key|
  x["#{key}_dup"] = 2
end
```
... which will raise an error.

## Expected behavior

Does not autocorrect (and lint maybe?) such code.

## Actual behavior

Auto-correction breaks ruby code.

## Steps to reproduce the problem

```shell
echo 'x = { a: 1 }; x.keys.each { |key| x["#{key}_dup"] = 2 }; puts x' > example.rb
ruby example.rb # works fine
rubocop -A example.rb
ruby example.rb # throws an error
```

## RuboCop version

```
$  rubocop -V
1.66.1 (using Parser 3.3.3.0, rubocop-ast 1.32.3, running on ruby 3.2.2) [x86_64-linux]
```
